### PR TITLE
feat: fixed placeholder position & auto-increase textarea

### LINKED ISSUE #302: [FEATURE] Auto increase message input height for multi-line messages
### Description

The message input field does not auto-expend when typing multi-line messages, which makes viewing long messages before sending them hard.

### Screenshots

_No response_

### Additional information

_No response_

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

### LINKED ISSUE #313: Wrong positioning of placeholder text
### Description

Placeholder for inputting message(s) in chat is positioned wrongly, it was not centered right.

### Screenshots

whisper img

### Additional information

_No response_

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by bintus-ux**
# Fixes Issue

**My PR closes #313 and closes #302 **

# 👨‍💻 Changes proposed(What did you do ?) 
- Adjusted placeholder text height correctly.
- Amended textarea multi-line messages to auto-increase and accomodate inputted texts.

# ✔️ Check List (Check all the applicable boxes)

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

Chat jsx - Whisper - Visual Studio Code 10_5_2023 11_32_1
![Chat jsx - Whisper - Visual Studio Code 10_5_2023 11_32_32 PM
6 PM](https://github.com/Dun-sin/Whisper/assets/122406759/31511646-be68-4412-b6bb-f4796f702ac7)
Whisper _ Chat Application - Google Chrome 10_5_2023 11_34_10 PM
80c10254d8e)

### CODE REVIEW

File: client/src/components/Chat.jsx [hunk 1]
```diff
@@ -589,13 +599,15 @@ const Chat = () => {
...
-                        ref={inputRef}
-                        onChange={handleTypingStatus}
-                    />
+                  <div className="w-full flex items-center justify-between bg-secondary rounded-l-md" id='message-input-container'>
+                <textarea
+                    placeholder="Send a Message..."
+                    id="message-input"
+                    className="h-[45px] focus:outline-none w-[96%] bg-secondary text-white rounded-[15px] resize-none pl-[22px] pr-[22px] py-[10px] text-[16px] placeholder-shown:align-middle"
+                    ref={inputRef}
+                    value={message}
```
Comment (Dun-sin) [inline client/src/components/Chat.jsx:608]: i don't think this concerns your assigned issue

File: client/src/components/Chat.jsx [hunk 2]
```diff
@@ -329,12 +330,21 @@ const Chat = () => {
 
     };
 
+    const adjustTextareaHeight = () => {
+        if (inputRef.current) {
+          inputRef.current.style.height = '45px';
```
Comment (Dun-sin) [inline client/src/components/Chat.jsx:335]: use a relative value instead

File: client/.eslintcache
Comment (Dun-sin) [inline client/.eslintcache:1]: remove this file

File: client/src/components/Chat.jsx [hunk 3]
```diff
@@ -40,6 +40,7 @@ const Chat = () => {
         messageID: null,
     });
     const [isQuoteReply, setIsQuoteReply] = useState(false)
+    const [message, setMessage] = useState('');
```
Comment (Dun-sin) [inline client/src/components/Chat.jsx:43]: this doesn't concern your issue either, unless you are using it to calculate the height, which i don't see

File: client/src/styles/chat.css [hunk 4]
```diff
@@ -66,6 +66,16 @@
...
+    .message-input-container {
+        max-height: 150px; 
+      }
+      
+      #message-input {
+        min-height: 40px; 
+        max-height: 100px; 
+        overflow-y: scroll;
+      }
+
```
Comment (Dun-sin) [inline client/src/styles/chat.css:78]: use tailwind for this

**Review by Dun-sin — CHANGES_REQUESTED** [review]

File: client/src/components/Chat.jsx (hunk 3 above)
Comment (bintus-ux) [reply client/src/components/Chat.jsx:43]: Yes it was used to calculate the text input  height through the message to setMessage(e.target.value)

File: client/src/styles/chat.css (hunk 4 above)
Comment (bintus-ux) [reply client/src/styles/chat.css:78]: It seemed not to work using tailwind but I would try again

File: client/src/components/Chat.jsx (hunk 2 above)
Comment (bintus-ux) [reply client/src/components/Chat.jsx:335]: Yeah I really had no idea where that file came from tried deleting it

File: client/.eslintcache
Comment (bintus-ux) [reply client/.eslintcache:1]: Yeah I had no idea where that file came from tried deleting it

File: client/src/components/Chat.jsx (hunk 1 above)
Comment (bintus-ux) [reply client/src/components/Chat.jsx:608]: I kinda needed to calculate the message inputted and adjusted the placeholder text position using py

File: client/src/components/Chat.jsx (hunk 2 above)
Comment (bintus-ux) [reply client/src/components/Chat.jsx]: @Dun-sin you can check it again now, just committed
Chat jsx - Whisper - Visual Studio Code 10_9_2023 10_33_39 AM

File: client/src/styles/chat.css (hunk 4 above)
Comment (bintus-ux) [reply client/src/styles/chat.css]: Sorted out
Chat jsx - Whisper - Visual Studio Code 10_9_2023 10_34_48 AM

**Review by bintus-ux — COMMENTED** [review]
Fixed the value issue and amended basic css styling to using tailwind css
Chat jsx - Whisper - Visual Studio Code 10_9_2023 10_33_39 AM
Chat jsx - Whisper - Visual Studio Code 10_9_2023 10_34_48 AM

**Review by Dun-sin — DISMISSED** [review]
You did great, thanks for contributing, I hope you will stick around and continue to contribute to this project.

Consider giving this project a star and joining the community discord server if you haven't for more resources and opportunities to connect with others. [👉🏽here👈🏽](https://discord.com/invite/ufcysW9q23)

**Comment by Dun-sin** [comment]
@bintus-ux you need to fix the conflicts

**Comment by bintus-ux** [comment]
> @bintus-ux you need to fix the conflicts

Yeah been confused about that bit with the eslintcache, how do I go about the file?

**Comment by Dun-sin** [comment]
> > @bintus-ux you need to fix the conflicts
> 
> Yeah been confused about that bit with the eslintcache, how do I go about the file?

this is why you should always do `git pull` before `git push` it was added to gitignore and shouldn't have been pushed

**Comment by bintus-ux** [comment]
> > > @bintus-ux you need to fix the conflicts
> > 
> > 
> > Yeah been confused about that bit with the eslintcache, how do I go about the file?
> 
> this is why you should always do `git pull` before `git push` it was added to gitignore and shouldn't have been pushed. it shouldn't be in your PR as an addition or substraction or deletion

Oh shit my bad sorry, forgot that. So I should pull and push and get that resolved then make a new PR?

**Comment by bintus-ux** [comment]
@Dun-sin please i've deleted the file and successfully merged with dev still can not get through the conflict, i need help.

**Pull request closed without merging**